app/blueprints/assessments/routes.py

```diff
@@ -949,6 +951,47 @@ def assignment_overview(fund_short_name: str, round_short_name: str):
 )
 @check_access_fund_short_name_round_sn
 def fund_dashboard(fund_short_name: str, round_short_name: str):
+    # Create a unique key for the session based on the fund short name and round short name
+    session_key = f"filter_params_{fund_short_name.upper()}_{round_short_name.upper()}"
+
+    # If there are no query arguments in the request
+    if not request.args:
+        # Check if there are saved filter parameters in the session
+        if session_key in session and session[session_key] is not None:
+            filter_params = session[session_key]
+            # If the 'clear_filters' flag is set, restore all saved filter parameters
+            if "clear_filters" in filter_params:
+                request.args = ImmutableMultiDict(session[session_key])
+            # Otherwise, merge saved filter parameters with existing query arguments
+            else:
+                request.args = ImmutableMultiDict(
+                    {
+                        **request.args,
+                        "search_term": filter_params.get(
+                            "search_term", request.args.get("search_term", "")
+                        ),
+                        "status": filter_params.get(
+                            "status", request.args.get("status", "All")
+                        ),
+                        "assigned_to": filter_params.get(
+                            "assigned_to", request.args.get("assigned_to", "All")
+                        ),
+                        "filter_by_tag": filter_params.get(
+                            "filter_by_tag", request.args.get("filter_by_tag", "ALL")
+                        ),
+                    }
+                )
+    # If the 'clear_filters' flag is set in the query arguments, save all query arguments to the session
+    elif "clear_filters" in request.args:
```

Comment:
I'm a bit confused here - if `clear_filters` is True, don't we want to clear the session data? (it looks like we're setting it instead : `session[session_key] = dict(request.args)` )

Reply:
Good catch, updated the code to clear session instead of saving them when clearing filters, thank you.